internal/core/data/router.go

```diff
@@ -72,3 +80,212 @@ func LoadRestRoutes() *mux.Router {
 
 	return r
 }
+
+/*
+Return number of events in Core Data
+/api/v1/event/count
+*/
+func eventCountHandler(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	switch r.Method {
+	case http.MethodGet:
+		count, err := count()
```

Comment:
- Having had the opportunity to check out the (implementation) details in this PR, I'm very pleased by the conceptual simplification offered by the approach. I like how the (seemingly inevitable) entanglement of business logic with the routing and plumbing (aka INFRA) is disentangled! 

In fact, I'm happy to report that I've now made use of this pattern already in [my System Management (SMA) work](https://github.com/akramtexas/edgex-go/commits/sys-management-agent).

Reply:
At least change ``count`` to just return ``dbClient.EventCount``.

I don't really understand how the code is decoupled just by calling ``count`` that only calls ``dbClient.EventCount``.
It makes all the sense to create ``countByDevice`` instead of adding the logic of checking the device existence to the http handler. But I don't agree with ``count``, but it is probably just personal style than a real concern.